packages/components/src/components/vertical-tab/vertical-tab.tsx

```diff
@@ -1,23 +1,63 @@
-import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
+import classNames from 'classnames';
 import { matchRoute } from '@deriv/shared';
-import VerticalTabContentContainer from './vertical-tab-content-container.jsx';
-import VerticalTabHeader from './vertical-tab-header.jsx';
-import VerticalTabHeaderGroup from './vertical-tab-header-group.jsx';
-import VerticalTabHeaders from './vertical-tab-headers.jsx';
-import VerticalTabHeaderTitle from './vertical-tab-header-title.jsx';
-import VerticalTabLayout from './vertical-tab-layout.jsx';
-import VerticalTabWrapper from './vertical-tab-wrapper.jsx';
+import VerticalTabContentContainer from './vertical-tab-content-container';
+import VerticalTabHeader, { TItem } from './vertical-tab-header';
+import VerticalTabHeaderGroup from './vertical-tab-header-group';
+import VerticalTabHeaders from './vertical-tab-headers';
+import VerticalTabHeaderTitle from './vertical-tab-header-title';
+import VerticalTabLayout from './vertical-tab-layout';
+import VerticalTabWrapper from './vertical-tab-wrapper';
 import Icon from '../icon/icon';
 
-const setSelectedIndex = ({ current_path, list, is_routed, selected_index, setCurrTabIndex, setVerticalTabIndex }) => {
-    let index;
+type TSetSelectedIndex = {
+    current_path: string;
+    list: TItem[];
+    is_routed?: boolean;
+    selected_index?: number | TItem | undefined;
```

Comment:
```suggestion
    selected_index?: number | TItem;
```

Reply:
thanks.. I missed it accidentally while experimenting :)